cfme/tests/infrastructure/test_vm_clone.py

```diff
@@ -0,0 +1,55 @@
+# -*- coding: utf-8 -*-
+import pytest
+
+from utils.providers import setup_a_provider as _setup_a_provider
+from cfme.infrastructure.virtual_machines import Vm
+from cfme.services import requests
+from utils.wait import wait_for
+from utils import testgen
+from utils.log import logger
+from utils.randomness import generate_random_string
+
+pytestmark = [
+    pytest.mark.fixtureconf(server_roles="+automate")
+]
+
+pytest_generate_tests = testgen.generate(testgen.provider_by_type, 'virtualcenter', scope="module")
+
+
+@pytest.fixture(scope="module")
+def setup_a_provider():
+    _setup_a_provider("infra")
+
+
+@pytest.fixture(scope="function")
+def vm_name():
+    vm_name = 'test_cloning_{}'.format(generate_random_string())
+    return vm_name
+
+
+def cleanup_vm(vm_name, provider_key, provider_mgmt):
+    try:
+        logger.info('Cleaning up VM %s on provider %s' % (vm_name, provider_key))
+        provider_mgmt.delete_vm(vm_name + "_0001")
+    except:
+        # The mgmt_sys classes raise Exception :\
+        logger.warning('Failed to clean up VM %s on provider %s' % (vm_name, provider_key))
+
+
+@pytest.mark.usefixtures("setup_a_provider")
+@pytest.mark.long_running
+def test_vm_clone(provider_type, provider_crud,
+                  provider_mgmt, request, vm_name, provider_key):
+    request.addfinalizer(lambda: cleanup_vm(vm_name, provider_key, provider_mgmt))
+    if provider_type == 'rhevm':
```

Comment:
these should be defined and pulled from cfme_data, please use better named to describe what these VMs are to the unfamiliar.  I am thinking it would be better to first provision the vm to clone as a pre-req for the test and then clone it.  I question whether muliple tests can clone the same VM at the same time which, if it is likely to error, we need to avoid.